rotkehlchen/api/rest.py

```diff
@@ -2730,21 +2710,55 @@ def delete_blockchain_transaction_data(
         return api_response(OK_RESULT, status_code=HTTPStatus.OK)
 
     @async_api_call()
-    def refresh_evm_transactions(
+    def refresh_transactions(
             self,
-            filter_query: EvmTransactionsFilterQuery,
+            from_timestamp: Timestamp,
+            to_timestamp: Timestamp,
+            accounts: list[BlockchainAccountData] | None,
     ) -> dict[str, Any]:
-        chain_ids: tuple[SUPPORTED_CHAIN_IDS]
-        if filter_query.chain_id is None:
-            chain_ids = get_args(SUPPORTED_CHAIN_IDS)
-        else:
-            chain_ids = (filter_query.chain_id,)
+        blockchain_addresses: dict[SupportedBlockchain, ListOfBlockchainAddresses]
+        with self.rotkehlchen.data.db.conn.read_ctx() as cursor:
+            if accounts is None or len(accounts) == 0:
+                blockchain_addresses = {
+                    chain: addr_list for chain in CHAINS_WITH_TRANSACTIONS
+                    if len(addr_list := self.rotkehlchen.data.db.get_single_blockchain_addresses(
+                        cursor=cursor,
+                        blockchain=chain,
+                    )) != 0
+                }
+            else:
+                blockchain_addresses = defaultdict(list)
+                for account in accounts:
+                    chains = (
+                        [account.chain] if account.chain is not None else
+                        self.rotkehlchen.data.db.get_account_blockchains(
+                            cursor=cursor,
+                            account=account.address,
+                        )
+                    )
+                    for chain in chains:
+                        blockchain_addresses[chain].append(account.address)  # type: ignore[arg-type]  # accounts with the same chain will have the same type of address
 
         result, message, status_code = True, '', HTTPStatus.OK
-        for chain_id in chain_ids:
-            evm_manager = self.rotkehlchen.chains_aggregator.get_evm_manager(chain_id)
+        for blockchain, addresses in blockchain_addresses.items():
             try:
-                evm_manager.transactions.query_chain(filter_query)
+                if blockchain.is_evm():
+                    self.rotkehlchen.chains_aggregator.get_evm_manager(
```

Comment:
you can use `get_chain_manager()` here as `self.rotkehlchen.chains_aggregator.get_evm_manager()` will convert the chain id back to `SupportedBlockchain`